### PR TITLE
add fallback for expected legacy default query revision functionality

### PR DESCRIPTION
**Pull request by scottvrosenthal**
add fallback for expected legacy default query revision functionality

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@ pkg
 *.patch
 *.gem
 *.sh
+.idea
```
Comment (felixbuenemann) [inline .gitignore]: This belongs in your [global gitignore file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

**Comment by felixbuenemann** [comment]
Related to #1467.

**Comment by scottvrosenthal** [comment]
@felixbuenemann updated PR based on comments.

**Comment by felixbuenemann** [comment]
Great, let's see what @leehambley thinks.

**Comment by leehambley** [comment]
I'll see if I can get a new release out soon.

**Comment by felixbuenemann** [comment]
Thanks, I can confirm that this fixes the regression reported in #1467.

**Comment by dideler** [comment]
I can't find v2.15.7 (introduced by this changeset) on RubyGems: https://rubygems.org/gems/capistrano/versions

Does this version still need to be released?

**Comment by leehambley** [comment]
There is no v2.15.7!

**Comment by leehambley** [comment]
My log shows we're overdue a release, but I don't have time to test one properly - maybe you can try master and report back:

```
*   5213ac6 - (HEAD -> legacy-v2, origin/legacy-v2) Merge pull request #1503 from bazzargh/patch-2 (5 weeks ago) <Lee Hambley>
|\
| * 180141c - fix rescue for asset_manifest_prefix (5 weeks ago) <bazzargh>
|/
*   07fefa9 - Merge pull request #1493 from kylerippey/changelog-typo-fix (7 weeks ago) <Lee Hambley>
|\
| * 1a6d865 - Fixed a minor typo in the CHANGELOG (7 weeks ago) <Kyle Rippey>
|/
*   34fe39c - Merge pull request #1481 from vsco/fix-svn-auth (3 months ago) <Lee Hambley>
|\
| * 6964046 - Fixes Subversion authentication arguments. (3 months ago) <Vlad Losev>
|/
*   b856989 - Merge pull request #1470 from activenetwork/legacy-v2 (3 months ago) <Lee Hambley>
|\
| * ffe4a23 - add fallback for expected legacy default query revision functionality (3 months ago) <scottvrosenthal>
|/
* dcd3257 - (tag: v2.15.6) Preparing Release v2.15.6 (5 months ago) <Lee Hambley>
* 265f3bf - fixed asset deploy problem for sprockets versions > 3.0 (5 months ago) <Stefan Kaes>
```

**Comment by dideler** [comment]
> but I don't have time to test one properly - maybe you can try master and report back:

@leehambley what testing is required for this project before tagging a new release? Is running the test suite sufficient or is there manual testing as well?

**Comment by leehambley** [comment]
I don't have any 2.x projects myself anymore, so if you are actively using the `HEAD` of the `legacy-v2` branch, the would-be target for 2.15.7, and can report that it's basically normal for you, that'd be one more (than the current 0) vote in favour of tagging and releasing it.

Releasing software that I haven't used in two years makes me nervous though. You really should upgrade to v3!

**Comment by felixbuenemann** [comment]
I have a couple of projects that still use capistrano 2.x that I could test against the legacy-v2 branch. I think a release to fix the regressions in 2.15.6 would be a good idea.

**Comment by dideler** [comment]
>  I think a release to fix the regressions in 2.15.6 would be a good idea.

This is mainly why I'm interested in 2.15.7. I'll try using `HEAD` of `legacy-v2` on a couple of our legacy apps that still use v2.x and will report back.

**Comment by felixbuenemann** [comment]
I've checked two projects that were locked to 2.15.5 due to the issues in 2.15.6, one using rails 3.2 and the other using rails 4.2 and they both deployed fine using 2.15.7 from the legacy-v2 branch.

**Comment by leehambley** [comment]
Thanks! I'm away on a workshop with a client. But I'll try and tag the
release today (late).

**Comment by leehambley** [comment]
Done (see #1526)

**Comment by dideler** [comment]
:tada:

**Pull request merged**